### PR TITLE
+doc #21668 better docs for configuring ssl

### PR DESCRIPTION
**Pull request by ktoso**
Refs https://github.com/akka/akka/issues/21668

Somewhat better already.

### CODE REVIEW

File: akka-docs/rst/scala/remoting.rst [hunk 1]
```diff
@@ -512,9 +506,45 @@ An example of setting up the default Netty based SSL driver as default::
...
+        random-number-generator = "AES128CounterSecureRNG"
+        enabled-algorithms = [TLS_RSA_WITH_AES_128_CBC_SHA]
+      }
+    }
+  }
+
+Creating and working with keystores and cerfiticates is well documented in the `Generating X.509 Certificates <http://typesafehub.github.io/ssl-config/CertificateGeneration.html#using-keytool>`_
+section of Lightbend's SSL-Config library. 
+
+Since an Akka remoting is inherently :ref:`peer-to-peer <symmetric-communication>` both the key-store as well as trust-store
```
Comment (patriknw) [inline akka-docs/rst/scala/remoting.rst]: remoting *node*

File: akka-docs/rst/scala/remoting.rst [hunk 2]
```diff
@@ -512,9 +506,45 @@ An example of setting up the default Netty based SSL driver as default::
...
+Creating and working with keystores and cerfiticates is well documented in the `Generating X.509 Certificates <http://typesafehub.github.io/ssl-config/CertificateGeneration.html#using-keytool>`_
+section of Lightbend's SSL-Config library. 
+
+Since an Akka remoting is inherently :ref:`peer-to-peer <symmetric-communication>` both the key-store as well as trust-store 
+need to be configured on each node participating in the cluster.
+
+The official `Java Secure Socket Extension documentation <http://docs.oracle.com/javase/7/docs/technotes/guides/security/jsse/JSSERefGuide.html>`_
+as well as the :ref:`Oracle documentation on creating KeyStore and TrustStores <https://docs.oracle.com/cd/E19509-01/820-3503/6nf1il6er/index.html>`
+are both great resources to research when setting up security on the JVM. Please consult those resources when troubleshooting
+and configuring SSL.
```
Comment (patriknw) [inline akka-docs/rst/scala/remoting.rst]: are there specific config values that we recommend, or recommend against?

Comment (ktoso) [inline akka-docs/rst/scala/remoting.rst]: I was researching this and found that the settings we have posted here are "OK", but I'm not sure what else could be a recommendation here. @wsargent do you have any hints what else we should suggest?

Comment (wsargent) [inline akka-docs/rst/scala/remoting.rst]: Also, the `Djava.security.egd=file:/dev/./urandom` note is no longer valid in JDK 1.8 and specifying file:/dev/urandom is fine.

File: akka-docs/rst/scala/remoting.rst [hunk 3]
```diff
@@ -512,9 +506,45 @@ An example of setting up the default Netty based SSL driver as default::
...
+
+See also a description of the settings in the :ref:`remote-configuration-scala` section.
 
 .. note::
```
Comment (wsargent) [inline akka-docs/rst/scala/remoting.rst]: Also, the Djava.security.egd=file:/dev/./urandom note is no longer valid in JDK 1.8 and specifying file:/dev/urandom is fine.

**Comment by gosubpl** [comment]
Please take a look at https://github.com/akka/akka/pull/21716#issuecomment-256770749

File: akka-docs/rst/scala/remoting.rst (hunk 1 above)
Comment (wsargent) [inline akka-docs/rst/scala/remoting.rst]: `cerfiticates` *certificates*

File: akka-docs/rst/java/remoting.rst [hunk 4]
```diff
@@ -480,42 +476,57 @@ untrusted mode when incoming via the remoting layer:
...
+
+Next the actual SSL/TLS parameters have to be configured::
 
+  akka {
+    remote {
       netty.ssl.security {
-        key-store = "mykeystore"
-        trust-store = "mytruststore"
+        key-store = "/example/path/to/mykeystore.jks"
+        trust-store = "/example/path/to/mytruststore.jks"
```
Comment (johanandren) [inline akka-docs/rst/java/remoting.rst]: 👍 I have seen people misunderstanding that more than a few times.

File: akka-docs/rst/java/remoting.rst [hunk 5]
```diff
@@ -480,42 +476,57 @@ untrusted mode when incoming via the remoting layer:
...
+        
         key-store-password = "changeme"
         key-password = "changeme"
         trust-store-password = "changeme"
+        
         protocol = "TLSv1.2"
+        
+        enabled-algorithms = [TLS_DHE_RSA_WITH_AES_128_GCM_SHA256]
+        
         random-number-generator = "AES128CounterSecureRNG"
```
Comment (johanandren) [inline akka-docs/rst/java/remoting.rst]: Shouldn't the default documented be SecureRandom based on what Will says and the discussions on gitter today?

**Review by johanandren — APPROVED** [review]
LGTM, we can always improve even more over time.

File: akka-docs/rst/scala/remoting.rst (hunk 2 above)
Comment (drewhk) [inline akka-docs/rst/scala/remoting.rst]: Seems like there is a performance impact of that still, compared to AES-CTR RNG.

File: akka-docs/rst/scala/remoting.rst (hunk 3 above)
Comment (drewhk) [inline akka-docs/rst/scala/remoting.rst]: That is true, but I would test this in practice on Jenkins.

**Comment by drewhk** [comment]
LGTM

File: akka-docs/rst/scala/remoting.rst (hunk 3 above)
Comment (drewhk) [inline akka-docs/rst/scala/remoting.rst]: I would still test this on Jenkins just to be sure.

File: akka-docs/rst/java/remoting.rst (hunk 5 above)
Comment (ktoso) [inline akka-docs/rst/java/remoting.rst]: yeah changing

Comment (gosubpl) [inline akka-docs/rst/java/remoting.rst]: :+1: to SecureRandom, others are performance optimizations IMHO

**Pull request merged**